server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -470,6 +472,25 @@ public boolean isRemoteStoreNode() {
         return this.getAttributes().keySet().stream().anyMatch(key -> key.startsWith(REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX));
     }
 
+    /**
+     * Returns whether the node is a remote cluster state enabled node.
+     * @return true if the node contains remote cluster state node attribute, false otherwise
+     */
+    public boolean isRemoteClusterStateEnabled() {
+        return this.getAttributes()
+            .keySet()
+            .stream()
+            .anyMatch(key -> (key.equals(REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY)));
+    }
+
+    /**
+     * Returns whether remote routing table is enabled on the node
+     * @return true if the node contains remote routing table node attributes, false otherwise
+     */
+    public boolean isRemoteRoutingTableEnabled() {
```

Comment:
multiple RoutingTable methods

Reply:
I see only `isRemoteRoutingTableEnabled`. Which is the other one?

Reply:
RemoteStoreNodeAttribute has same method defined as
```
   public static boolean isRemoteRoutingTableEnabled(Settings settings) {
        return FeatureFlags.isEnabled(REMOTE_PUBLICATION_EXPERIMENTAL) && isRemoteRoutingTableAttributePresent(settings);
    }
```

Reply:
`RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled` checks for publication feature flag and routing table node attributes. This is used to initialize `RemoteRoutingTableService`.
`DiscoveryNode.isRemoteRoutingTableEnabled` checks for routing table node attributes for follower nodes since we cannot check feature flag on remote nodes. This check along with `DiscoveryNode.isRemoteClusterStateEnabled` is used to decide if we can send the remote state to follower node.

Reply:
same for isRemoteStatePublicationEnabled as well defined in 2 places

Reply:
Removed from RemoteStoreNodeAttribute